pydicom/dataelem.py

```diff
@@ -535,8 +539,15 @@ def __eq__(self, other):
             return True
 
         if isinstance(other, self.__class__):
-            return (self.tag == other.tag and self.VR == other.VR
-                    and self.value == other.value)
+            if self.tag != other.tag or self.VR != other.VR:
+                return False
+
+            # tag and VR match, now check the value
+            if config.have_numpy and isinstance(self.value, numpy.ndarray):
+                return (len(self.value) == len(other.value)
+                        and numpy.allclose(self.value, other.value))
```

Comment:
Why `allclose` rather than `array_equal`? Also, the default tolerances might not be appropriate

Reply:
In my experience `allclose` works better when swapping between environments i.e. Linux and Mac. I had tests in my own library pass on my home PC but fail on Travis CI due to this.

Reply:
`array_equal` may be fine for IS values though.